libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -1106,11 +1106,22 @@ void CompilerUtils::convertType(
 			}
 			case DataLocation::CallData:
 			{
-				solUnimplementedAssert(!typeOnStack.isDynamicallyEncoded(), "");
-				m_context << Instruction::DUP1;
-				m_context << Instruction::CALLDATASIZE;
-				m_context << Instruction::SUB;
-				abiDecode({&targetType}, false);
+				if (typeOnStack.isDynamicallyEncoded())
```

Comment:
Actually, `bytes` does not need abi v2 - right? It has to be an array with a dynamically encoded base type.

Reply:
True about bytes. I am now confused why some tests didn't fail. Probably they have pragma for coder v2...
Will fix this now

Reply:
Actually, I added test that copies `struct` containing `bytes` array and it needs `pragma abicoder v2`.
So, this just implements when type is struct. I'll check missing conversions for arrays and will activate them in separate PR.

Reply:
Ah right! Yes, structs always require v2